packages/dm-core-plugins/src/generic-list/GenericTablePlugin.tsx

```diff
@@ -0,0 +1,190 @@
+          onClick={() => addItem(idReference, 'type')}
+        >
+          <Icon data={library_add} title="Append" />
+        </Button>
+      </div>
+      <Table style={{ width: '100%' }}>
+        <Table.Head>
+          <Table.Row>
+            {internalConfig.columns.map((attribute: string) => (
+              <Table.Cell key={attribute}>{attribute}</Table.Cell>
+            ))}
+            {internalConfig.editMode && internalConfig.showDelete && (
+              <Table.Cell>Delete</Table.Cell>
+            )}
+          </Table.Row>
+        </Table.Head>
+        <Table.Body>
+          {Object.entries(items).map(([key, item], index) => (
+            <Table.Row key={key}>
+              {internalConfig.columns.map((attribute: string) => {
+                if (typeof item[attribute] === 'object')
+                  throw new Error(
+                    `Objects can not be displayed in table header. Attribute '${attribute}' is not a primitive type.`
+                  )
+                // TODO: Consider having a more robust way of getting type
+                const attributeType = typeof item[attribute]
+                return (
+                  <Table.Cell key={attribute}>
+                    <Input
+                      value={item[attribute] || ''}
+                      readOnly={
+                        attribute === 'type' || !internalConfig.editMode
+                      }
+                      type={attributeType}
+                      rightAdornments={
+                        <>{internalConfig.editMode ? attributeType : ''}</>
+                      }
+                      onChange={(event: ChangeEvent<HTMLInputElement>) => {
+                        let newValue: string | number | boolean =
+                          event.target.value
+                        if (attributeType === 'number')
+                          newValue = Number(newValue)
+                        updateItem(key, attribute, newValue)
+                      }}
+                    />
+                  </Table.Cell>
+                )
+              })}
+
+              {internalConfig.editMode && internalConfig?.showDelete && (
+                <Table.Cell style={{ textAlign: 'right' }}>
+                  <Button
+                    variant="ghost_icon"
+                    color="danger"
+                    onClick={() => deleteItem(`${idReference}.${index}`, key)}
+                  >
+                    <Icon data={delete_to_trash} title="Delete" />
+                  </Button>
+                </Table.Cell>
+              )}
+            </Table.Row>
+          ))}
+        </Table.Body>
+      </Table>
+      <div
+        style={{
+          display: 'flex',
+          flexDirection: 'row-reverse',
+          margin: '10px',
+        }}
+      >
+        <Button
+          variant="ghost_icon"
+          onClick={() => addItem(idReference, 'type')}
+        >
+          <Icon data={library_add} title="Append" />
+        </Button>
```

Comment:
This is written two places. Refactor?

Reply:
Created separate "AppendButton" component